### PR TITLE
Reagent container spawning refactor

### PR DESCRIPTION
**Pull request by Jordie0608**
See #6889

Reagent_containers handle spawning with add_reagent_list and list_reagent to support containers that spawn with multiple reagents. Any reagent_container can now also contain diseased blood of a variable (default 20) amount. The list_reagent var can safely be modified when mapping.

bottles.dm moved to parent directory as it was the only file in its directory.

Vestigial comments have been removed from some files in addition to some reorganization.

Reagent values of various foods have changed insignificantly for standardization's sake.
- [x] Framework
- [x] Chemistry and medical items (pills, syringes, bottles etc.)
- [x] Foods
- [x] Drinks

### CODE REVIEW

File: code/modules/food&drinks/food/condiment.dm [hunk 1]
```diff
@@ -127,16 +127,10 @@
...
-	..()
-	reagents.add_reagent("enzyme", 50)
+	list_reagents = list("enzyme" = 50)
 
 /obj/item/weapon/reagent_containers/food/condiment/sugar
-
-/obj/item/weapon/reagent_containers/food/condiment/sugar/New()
-	..()
-	reagents.add_reagent("sugar", 50)
+	list_reagents = list("SUGAR" = 50)
```
Comment (tkdrg) [inline code/modules/food&drinks/food/condiment.dm]: Is this supposed to be allcaps?

File: code/modules/food&drinks/food/snacks.dm [hunk 2]
```diff
@@ -255,21 +255,15 @@
 	icon_state = "cheesewheel"
 	slice_path = /obj/item/weapon/reagent_containers/food/snacks/cheesewedge
 	slices_num = 5
-
-/obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel/New()
-	..()
-	reagents.add_reagent("nutriment", 20)
-	reagents.add_reagent("vitamin", 5)
+	list_reagents = list("nutriment" = 20, "vitamin" = 5)
 	bitesize = 2
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks.dm]: This bitesize can go

File: code/modules/food&drinks/food/snacks_bread.dm [hunk 3]
```diff
@@ -130,29 +102,19 @@
...
 	desc = "Bon appetit!"
 	icon_state = "baguette"
-
-/obj/item/weapon/reagent_containers/food/snacks/baguette/New()
-	..()
-	reagents.add_reagent("nutriment", 6)
-	reagents.add_reagent("blackpepper", 1)
-	reagents.add_reagent("sodiumchloride", 1)
-	reagents.add_reagent("vitamin", 1)
+	list_reagents = list("nutriment" = 60, "blackpepper" = 1, "sodiumchloride" = 1, "vitamin" = 1)
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_bread.dm]: 60? That's a lot.

File: code/modules/food&drinks/food/snacks_meat.dm [hunk 4]
```diff
@@ -8,25 +8,18 @@
...
 
 /obj/item/weapon/reagent_containers/food/snacks/carpmeat
 	name = "carp fillet"
 	desc = "A fillet of spess carp meat."
 	icon_state = "fishfillet"
+	list_reagents = list("nutriment" = 2, "carpotoxin" = 2, "vitamin" = 2)
+	bitesize = 6
 
 /obj/item/weapon/reagent_containers/food/snacks/carpmeat/New()
-	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_meat.dm]: Don't remove ..() from a New().

File: code/modules/food&drinks/food/snacks_meat.dm [hunk 5]
```diff
@@ -272,80 +183,39 @@
 	desc = "A nice and warm stew. Healthy and strong."
 	icon_state = "stew"
 	trash = /obj/item/trash/snack_bowl
+	list_reagents = list("nutriment" = 10, "tomatojuice" = 5, "oculine" = 5, "water" = 5)
+	bitesize = 10
 
 /obj/item/weapon/reagent_containers/food/snacks/stew/New()
-	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_meat.dm]: Don't remove ..()

File: code/modules/food&drinks/food/snacks_meat.dm [hunk 6]
```diff
@@ -58,162 +44,95 @@
...
+	list_reagents = list("nutriment" = 1, "vitamin" = 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/sausage
 	name = "sausage"
 	desc = "A piece of mixed, long meat."
 	icon_state = "sausage"
+	list_reagents = list("nutriment" = 5, "vitamin" = 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/sausage/New()
-	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_meat.dm]: Don't remove ..() here

File: code/modules/food&drinks/food/snacks_meat.dm [hunk 7]
```diff
@@ -272,80 +183,39 @@
...
 
 /obj/item/weapon/reagent_containers/food/snacks/stewedsoymeat
 	name = "stewed soy meat"
 	desc = "Even non-vegetarians will LOVE this!"
 	icon_state = "stewedsoymeat"
 	trash = /obj/item/trash/plate
+	list_reagents = list("nutriment" = 8)
 
 /obj/item/weapon/reagent_containers/food/snacks/stewedsoymeat/New()
-	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_meat.dm]: don't remove ..()

File: code/modules/food&drinks/food/snacks_other.dm [hunk 8]
```diff
@@ -48,209 +36,115 @@
...
 	name = "popcorn"
 	desc = "Now let's find some cinema."
 	icon_state = "popcorn"
 	trash = /obj/item/trash/popcorn
-	var/unpopped = 0
+	list_reagents = list("nutriment" = 2)
+	bitesize = 0.1 //this snack is supposed to be eating during looooong time. And this it not dinner food! --rastaf0
 
 /obj/item/weapon/reagent_containers/food/snacks/popcorn/New()
-	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_other.dm]: Please don't remove ..()

File: code/modules/food&drinks/food/snacks_pastry.dm [hunk 9]
```diff
@@ -1,115 +1,62 @@
...
-	src.bitesize = 3
+/obj/item/weapon/reagent_containers/food/snacks/donut/New()
 	if(prob(30))
-		src.icon_state = "donut2"
-		src.name = "frosted donut"
-		reagents.add_reagent("sprinkles", 2)
+		icon_state = "donut2"
+		name = "frosted donut"
+		list_reagents.Add("sprinkles" = 2)
+	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_pastry.dm]: It's good practice to call ..() first

File: code/modules/food&drinks/food/snacks_pastry.dm [hunk 10]
```diff
@@ -1,115 +1,62 @@
...
-			reagents.add_reagent("omnizine", 3)
+	extra_reagent = pick("nutriment", "capsaicin", "frostoil", "krokodil", "plasma", "coco", "slimejelly", "banana", "berryjuice", "omnizine")
+	list_reagents.Add(extra_reagent = 3)
 	if(prob(30))
 		icon_state = "donut2"
 		name = "frosted chaos donut"
-		reagents.add_reagent("sprinkles", 2)
-
+		list_reagents.Add("sprinkles" = 2)
+	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_pastry.dm]: Same here, ..() first

File: code/modules/food&drinks/food/snacks_pastry.dm [hunk 11]
```diff
@@ -1,115 +1,62 @@
...
-	reagents.add_reagent("berryjuice", 5)
-	reagents.add_reagent("vitamin", 1)
+	list_reagents.Add(extra_reagent = 5)
 	if(prob(30))
 		icon_state = "jdonut2"
-		name = "Frosted Jelly Donut"
-		reagents.add_reagent("sprinkles", 2)
+		name = "frosted jelly Donut"
+		list_reagents.Add("sprinkles" = 2)
+	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_pastry.dm]: ..() first

File: code/modules/food&drinks/food/snacks_pie.dm [hunk 12]
```diff
@@ -23,100 +17,59 @@
...
 	..()
 	if(prob(10))
 		name = "exceptional plump pie"
 		desc = "Microwave is taken by a fey mood! It has cooked an exceptional plump pie!"
-		reagents.add_reagent("omnizine", 5)
-
-	reagents.add_reagent("nutriment", 8)
-	reagents.add_reagent("vitamin", 4)
-	bitesize = 2
+		list_reagents.Add("omnizine" = 5)
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_pie.dm]: Don't add to list_reagents

File: code/modules/food&drinks/food/snacks_pastry.dm [hunk 13]
```diff
@@ -201,122 +121,75 @@
...
 	..()
 	if(prob(10))
 		name = "exceptional plump helmet biscuit"
 		desc = "Microwave is taken by a fey mood! It has cooked an exceptional plump helmet biscuit!"
-		reagents.add_reagent("nutriment", 3)
-		reagents.add_reagent("omnizine", 5)
-	reagents.add_reagent("nutriment", 5)
-	reagents.add_reagent("vitamin", 1)
-	bitesize = 2
+		list_reagents.Add("omnizine" = 5)
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_pastry.dm]: Adding to list_reagents won't do anything. use reagents.add_reagent instead.

File: code/modules/food&drinks/food/snacks_salad.dm [hunk 14]
```diff
@@ -1,41 +1,26 @@
 //this category is very little but I think that it has great potential to grow
 ////////////////////////////////////////////SALAD////////////////////////////////////////////
+/obj/item/weapon/reagent_containers/food/snacks/salad
+	trash = /obj/item/trash/snack_bowl
+	bitesize = 3
+
+/obj/item/weapon/reagent_containers/food/snacks/salad/New()
+	eatverb = pick("crunch","devour","nibble","gnaw","gobble","chomp")
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_salad.dm]: Don't make a New() without calling ..()

File: code/modules/food&drinks/food/snacks_soup.dm [hunk 15]
```diff
@@ -1,247 +1,111 @@
...
 	icon_state = "mysterysoup"
-	trash = /obj/item/trash/snack_bowl
-
-/obj/item/weapon/reagent_containers/food/snacks/mysterysoup/New()
+	list_reagents = list("nutriment" = 6, "tomatojuice" = 2)
+	var/extra_reagent = null
+/obj/item/weapon/reagent_containers/food/snacks/soup/mysterysoup/New()
+	extra_reagent = pick("capsaicin", "frostoil", "omnizine", "banana", "blood", "slimejelly", "toxin", "banana", "carbon", "oculine")
+	list_reagents.Add(extra_reagent = 5)
 	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_soup.dm]: list_reagents.Add, ..() should be first, etc

File: code/modules/food&drinks/food/snacks_soup.dm [hunk 16]
```diff
@@ -1,247 +1,111 @@
...
-	bitesize = 5
+/obj/item/weapon/reagent_containers/food/snacks/soup/wishsoup/New()
 	if(prob(25))
 		desc = "A wish come true!"
-		reagents.add_reagent("nutriment", 8)
-		reagents.add_reagent("vitamin", 1)
+		list_reagents = list("nutriment" = 9, "vitamin" = 1)
+	else
+		list_reagents = list("water" = 10)
+	..()
```
Comment (tkdrg) [inline code/modules/food&drinks/food/snacks_soup.dm]: list_reagents ..() etc

File: code/modules/reagents/reagent_containers/spray.dm [hunk 17]
```diff
@@ -214,8 +198,4 @@
 	icon_state = "plantbgone"
 	item_state = "plantbgone"
 	volume = 100
-
-
-/obj/item/weapon/reagent_containers/spray/plantbgone/New()
-	..()
-	reagents.add_reagent("plantbgone", 100)
+	list_reagents = list("plantbgone" = 30)
```
Comment (tkdrg) [inline code/modules/reagents/reagent_containers/spray.dm]: Is this reduction intentional?

**Comment by Jordie0608** [comment]
Master borked =|

**Comment by tkdrg** [comment]
Conflicts

**Comment by Jordie0608** [comment]
The conflicts just won't stop

**Pull request merged**